src/main/java/com/nike/vault/client/VaultClientFactory.java

```diff
@@ -85,11 +74,28 @@ public static VaultClient getClient(final UrlResolver vaultUrlResolver) {
      */
     public static VaultClient getClient(final UrlResolver vaultUrlResolver,
                                         final VaultCredentialsProvider vaultCredentialsProvider) {
-        return new VaultClient(vaultUrlResolver, vaultCredentialsProvider, new OkHttpClient.Builder()
-                .connectTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
-                .writeTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
-                .readTimeout(DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_UNIT)
-                .build());
+        return getClient(vaultUrlResolver, vaultCredentialsProvider, new Headers.Builder().build());
+    }
+
+    /**
+     * Factory method that allows a user to define default HTTP headers to be added to every HTTP request made from the
+     * VaultClient. The user can also define their Vault URL resolver and credentials provider.
+     *
+     * @param vaultUrlResolver    URL resolver for Vault
+     * @param vaultCredentialsProvider Credential provider for acquiring a token for interacting with Vault
+     * @return Vault client
+     */
+    public static VaultClient getClient(final UrlResolver vaultUrlResolver,
+                                        final VaultCredentialsProvider vaultCredentialsProvider,
+                                        final Headers defaultHeaders) {
```

Comment:
We should use a Map here for headers so that we don't expose the underlying library we're using.